python/ray/serve/config.py

```diff
@@ -241,6 +241,7 @@ class HTTPOptions(pydantic.BaseModel):
     port: int = DEFAULT_HTTP_PORT
     middlewares: List[Any] = []
     location: Optional[DeploymentMode] = DeploymentMode.HeadOnly
+    dedicated_cpu: bool = False
```

Comment:
let's just make this `num_cpus` because we may want to support multi-threaded servers in the future. `resources` dict might also be worthwhile but not sure if we'll ever need that level of expressiveness